test/mocha/20-resources.js

```diff
@@ -553,4 +554,84 @@ describe('resources', function() {
     const expectedDiff = 1;
     diff.should.equal(expectedDiff);
   });
+
+  it('should acquire successfully after expiration', async function() {
+    // use local `acquirerId` so uninfluenced by previous acquisitions
+    const acquirerId = uuid();
+    const seconds = 2;
+
+    const id = await generateId();
+    await restrictions.insert({
+      restriction: {
+        id,
+        zone: ZONES.ONE,
+        resource: RESOURCES.PLUM,
+        method: 'limitOverDuration',
+        methodOptions: {
+          limit: 1,
+          duration: `PT${seconds}S`
+        }
+      }
+    });
+
+    // acquire first resource
+    {
+      const now = Date.now();
+      const request = [
+        {resource: RESOURCES.PLUM, count: 1, requested: now}
+      ];
+      const acquisitionTtl = 30000;
+      const zones = [ZONES.ONE, ZONES.TWO];
+      const result = await resources.acquire(
+        {acquirerId, request, acquisitionTtl, zones});
+      const expectedResult = {
+        authorized: true,
+        excessResources: [],
+        untrackedResources: []
+      };
+      assertCheckResult(result, expectedResult);
+    }
+
+    // fail to acquire second resource
+    {
+      const now = Date.now();
+      const request = [
+        {resource: RESOURCES.PLUM, count: 1, requested: now}
+      ];
+      const acquisitionTtl = 30000;
+      const zones = [ZONES.ONE, ZONES.TWO];
+      const result = await resources.acquire(
+        {acquirerId, request, acquisitionTtl, zones});
+      const expectedResult = {
+        authorized: false,
+        excessResources: [{
+          resource: RESOURCES.PLUM,
+          count: 1
+        }],
+        untrackedResources: []
+      };
+      assertCheckResult(result, expectedResult);
+    }
+
+    // wait for expiration period
+    await new Promise(r => setTimeout(r, seconds * 1000));
```

Comment:
please use `bedrock.util.delay`.
```suggestion
    await delay(seconds * 1000);
```
https://www.npmjs.com/package/delay

Reply:
Updated.